### DISCUSSION TITLE
why @Schema(anyOf = {xxx.class, xxx.class}) doesn't work but @Schema(allOf = {xxx.class, xxx.class} can?)

### DISCUSSION THREAD

**Discussion by manchuanxingmengyaxinhe** [victools/jsonschema-generator#265] [Q&A] [answered]
this is SchemaGeneratorConfigBuilder
![image](https://user-images.githubusercontent.com/109338032/180195743-28887d08-b115-4e2c-bf33-6136aa1b43c2.png)
 this is Enity with @Schema
![image](https://user-images.githubusercontent.com/109338032/180196055-d9a003a9-3a62-4bc3-acc8-2e10fbcd4f11.png)
this is when @Schema user allOf Attributes the schema is
![image](https://user-images.githubusercontent.com/109338032/180196641-32e28e66-720b-47a2-82b9-02d9e7cbe3ba.png)

but when i change the @schema attributes to anyOf the Schema is 
![image](https://user-images.githubusercontent.com/109338032/180196965-857c37b2-4659-4a10-982d-fcbd44cdcee4.png)

thanks for you help. Sorry that I am a newbie

**Comment by CarstenWickner** [comment, marked as answer]
Hi @manchuanxingmengyaxinhe,

Short answer: Your observation is correct, while `allOf` is supported out-of-the-box, `anyOf` (or `oneOf` for that matter) is not handled by the Swagger 2.0 Module yet.
https://github.com/victools/jsonschema-generator/blob/08c724985562defec4edd4a07c97b97a58d006ba/jsonschema-module-swagger-2/src/main/java/com/github/victools/jsonschema/module/swagger2/Swagger2Module.java#L434-L439

I've created PR #266 now, to add that missing support accordingly.

Generally, a question like this is much easier to act upon if provided in textual form and not as screenshots.
For the next time then. 😉

**Reply by manchuanxingmengyaxinhe** [reply]
thanks you for the  answer !

**Reply by manchuanxingmengyaxinhe** [reply]
the next time i will provid in textual . thanks very much !
